js/src/toast/toast.spec.js

```diff
@@ -0,0 +1,222 @@
+        expect(toastEl.classList.contains('show')).toEqual(false)
+        done()
+      })
+
+      toast.show()
+    })
+
+    it('should not add fade class', done => {
+      fixtureEl.innerHTML = [
+        '<div class="toast" data-delay="1" data-animation="false">',
+        '  <div class="toast-body">',
+        '    a simple toast',
+        '  </div>',
+        '</div>'
+      ].join('')
+
+      const toastEl = fixtureEl.querySelector('.toast')
+      const toast = new Toast(toastEl)
+
+      toastEl.addEventListener('shown.bs.toast', () => {
+        expect(toastEl.classList.contains('fade')).toEqual(false)
+        done()
+      })
+
+      toast.show()
+    })
+  })
+
+  describe('hide', () => {
+    it('should allow to hide toast manually', done => {
+      fixtureEl.innerHTML = [
+        '<div class="toast" data-delay="1" data-autohide="false">',
+        '  <div class="toast-body">',
+        '    a simple toast',
+        '  </div>',
+        '  </div>'
+      ].join('')
+
+      const toastEl = fixtureEl.querySelector('.toast')
+      const toast = new Toast(toastEl)
+
+      toastEl.addEventListener('shown.bs.toast', () => {
+        toast.hide()
+      })
+
+      toastEl.addEventListener('hidden.bs.toast', () => {
+        expect(toastEl.classList.contains('show')).toEqual(false)
+        done()
+      })
+
+      toast.show()
+    })
+
+    it('should do nothing when we call hide on a non shown toast', () => {
+      fixtureEl.innerHTML = '<div></div>'
+
+      const toastEl = fixtureEl.querySelector('div')
+      const toast = new Toast(toastEl)
+
+      spyOn(toastEl.classList, 'contains')
+
+      toast.hide()
+
+      expect(toastEl.classList.contains).toHaveBeenCalled()
+    })
+  })
+
+  describe('dispose', () => {
+    it('should allow to destroy toast', () => {
+      fixtureEl.innerHTML = '<div></div>'
+
+      const toastEl = fixtureEl.querySelector('div')
+      const toast = new Toast(toastEl)
+
+      expect(Toast._getInstance(toastEl)).toBeDefined()
+
+      toast.dispose()
+
+      expect(Toast._getInstance(toastEl)).toBeNull()
+    })
+
+    it('should allow to destroy toast and hide it before that', done => {
+      fixtureEl.innerHTML = [
+        '<div class="toast" data-delay="0" data-autohide="false">',
+        '  <div class="toast-body">',
+        '    a simple toast',
+        '  </div>',
+        '</div>'
+      ].join('')
+
+      const toastEl = fixtureEl.querySelector('div')
+      const toast = new Toast(toastEl)
+
+      EventHandler.one(toastEl, 'shown.bs.toast', () => {
+        setTimeout(() => {
+          expect(toastEl.classList.contains('show')).toEqual(true)
+          expect(Toast._getInstance(toastEl)).toBeDefined()
+
+          toast.dispose()
+
+          expect(Toast._getInstance(toastEl)).toBeNull()
+          expect(toastEl.classList.contains('show')).toEqual(false)
+
+          done()
+        }, 1)
+      })
+      toast.show()
+    })
+  })
+
+  describe('config', () => {
```

Comment:
config isn't a public method, you should move those tests in `constructor` describe